### PR TITLE
Add template for testing out of tree cloud provider with Windows 

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**

**What this PR does / why we need it**: Add a template for https://github.com/kubernetes-sigs/cloud-provider-azure/issues/705 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**: This template has both a Linux pool and a Windows pool. I'm not sure how the CCM test suite handles testing LBs for Windows nodes or if it does at all. @nilo19 any recommentations? cc @jsturtevant 

/hold

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: templates/flavors/external-cloud-provider-windows/cloud-controller-manager.yaml [hunk 1]
```diff
@@ -0,0 +1,198 @@
+apiVersion: v1
```
Comment (jsturtevant) [inline templates/flavors/external-cloud-provider-windows/cloud-controller-manager.yaml:1]: should these files come from the cloud manage repository so any changes made to them would be reflected in the tests?

Comment (CecileRobertMichon) [reply templates/flavors/external-cloud-provider-windows/cloud-controller-manager.yaml:1]: we have some slight customizations in the cloud-controller-manager container, eg. image and clusterName. Do you know if kustomize let's you take a file url as base?

Side note, this should not have been duplicated in this PR though, I copied it into the new directory by accident, it should use the same file as we were using before.

Comment (jsturtevant) [reply templates/flavors/external-cloud-provider-windows/cloud-controller-manager.yaml]: This is an example of using remote resources: https://github.com/kubernetes-sigs/kustomize/blob/master/examples/loadHttp.md. Admittedly haven't tried it my self

Comment (feiskyer) [reply templates/flavors/external-cloud-provider-windows/cloud-controller-manager.yaml]: have someone confirmed whether the remote resources work or not?

Comment (CecileRobertMichon) [reply templates/flavors/external-cloud-provider-windows/cloud-controller-manager.yaml]: @jsturtevant I'm using the files in a ConfigMapGenerator, not as a resource, so I'm not sure I can use the remove resource there

**Pull request closed without merging**